packages/plugin-react-native-client-sync/test/client-sync.test.ts

```diff
@@ -107,6 +107,78 @@ describe('plugin: react native client sync', () => {
       })
       c.leaveBreadcrumb('Spin', { direction: 'ccw', deg: '90' })
     })
+
+    describe('feature flags', () => {
+      it('adds individual feature flags', done => {
+        const c = new Client({
+          apiKey: 'api_key',
+          plugins: [
+            plugin({
+              addFeatureFlag: (name: string, variant?: string) => {
+                expect(name).toBe('feature flag')
+                expect(variant).toBe('flag variant')
+                done()
+              }
+            })
+          ]
+        })
+
+        c.addFeatureFlag('feature flag', 'flag variant')
+      })
+
+      it('adds arrays of feature flags', done => {
+        const c = new Client({
+          apiKey: 'api_key',
+          plugins: [
+            plugin({
+              addFeatureFlags: (flags: { name: string, variant?: string }) => {
+                expect(flags).toStrictEqual([
+                  { name: 'feature flag', variant: 'flag variant' },
+                  { name: 'name only flag' }
+                ])
+                done()
+              }
+            })
+          ]
+        })
+
+        c.addFeatureFlags([
+          { name: 'feature flag', variant: 'flag variant' },
+          { name: 'name only flag' }
+        ])
+      })
+
+      it('clears specific feature flags', done => {
+        const c = new Client({
+          apiKey: 'api_key',
+          plugins: [
+            plugin({
+              clearFeatureFlag: (name: string) => {
+                expect(name).toStrictEqual('feature flag')
+                done()
+              }
+            })
+          ]
+        })
+
+        c.clearFeatureFlag('feature flag')
+      })
+
+      it('clears all feature flags', done => {
+        const c = new Client({
+          apiKey: 'api_key',
+          plugins: [
+            plugin({
+              clearFeatureFlags: () => {
+                done()
+              }
+            })
+          ]
+        })
+
+        c.clearFeatureFlags()
+      })
+    })
```

Comment:
ESLint doesn't like this because it doesn't technically have an assertion 😞

Since it's all synchronous, this could use a mock instead:

```js
it('clears all feature flags', () => {
  const clearFeatureFlags = jest.fn()
  const c = new Client({
    apiKey: 'api_key',
    plugins: [plugin({ clearFeatureFlags })]
  })

  c.clearFeatureFlags()
  expect(clearFeatureFlags).toHaveBeenCalledTimes(1)
})
```